vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -664,7 +664,13 @@ def add_remote_agent(self,
         # TODO re-evaluate refreshing for scaling/recovery
         if remote_tp_rank in self._remote_agents.get(engine_id, ()):
             return
-
+        if remote_tp_rank != self.tp_rank:
```

Comment:
The root cause lies here: 
https://github.com/vllm-project/vllm/blob/9ef9173cfa338ea7742f23ee9d0038e580081390/vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py#L470-L473

when the process with rank 7 is retrieving the metadata, it incorrectly records the `num_blocks`.
https://github.com/vllm-project/vllm/blob/9ef9173cfa338ea7742f23ee9d0038e580081390/vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py#L708-L712
Specifically, rank 7 of D records the `num_blocks` from rank 0 of P.

Reply:
this won't work when local TP > remote TP

Reply:
@NickLucche I have updated the calculation method. Please take a look again.